src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/WaySequence.java

```diff
@@ -0,0 +1,84 @@
+package org.openstreetmap.josm.plugins.pt_assistant.data;
+
+import org.openstreetmap.josm.data.osm.Relation;
+import org.openstreetmap.josm.data.osm.RelationMember;
+import org.openstreetmap.josm.data.osm.Way;
+
+import static org.openstreetmap.josm.plugins.pt_assistant.utils.RouteUtils.isPTWay;
+
+public final class WaySequence<previousWay extends Way, currentWay extends Way, nextWay extends Way, wayAfterNextWay extends Way> {
+    public Way previousWay;
+    public Way currentWay;
+    public Way nextWay;
+    public Way wayAfterNextWay;
```

Comment:
For such a tuple class it would be useful, if the fields were immutable (i.e. `final`), so they don't change after creation and especially can't become `null` unexpectedly.

Reply:
If I make them final, my setters don't work anymore. Also I almost expect them to be null every once in a while (when currentWay is the first, last or before last highway in the route relation). currentWay should never be null however. Actually it could be null, when the stops are reached (going backwards). I changed that now. (but not committed yet)

Reply:
Final fields could still be null, the useful thing about them is, that during the lifetime of an object they can't change. So they can't be non-null and then later become null unexpectedly.
Yes, final fields can't have setters, all fields have to be set through the constructor.